echopype/utils/io.py

```diff
@@ -280,3 +280,30 @@ def check_file_permissions(FILE_DIR):
                 TEST_FILE.unlink()
     except Exception:
         raise PermissionError("Writing to specified path is not permitted.")
+
+
+def join_paths(*args: Tuple[str, ...]) -> str:
```

Comment:
Currently this function is only used in `zarr_combine._modify_prov_filename()` and the test functions, and not anywhere else. The functionality is among those provided by `pathlib`. I suggest removing this function and adding a block in the existing function `validate_output_path` to handle the Windows path changes -- this seems a good way to ensure that the path differences are handled correctly everywhere.

Reply:
Based on prior discussions we will not remove `join_paths` and we will proceed with the consensus established in the [PR comment](https://github.com/OSOceanAcoustics/echopype/pull/879#issuecomment-1320580649).